### PR TITLE
Optionally set mariadb securityContext

### PR DESCRIPTION
**Pull request by djzager**
**What this PR does / why we need it**:

This makes it possible to disable the `securityContext` settings when using mariadb, for example: when installing the chart in OpenShift.

**Special notes for your reviewer**:

N/A

### CODE REVIEW

**Comment by djzager** [comment]
Manual Testing:

```
$  helm install ./mariadb-2.1.15.tgz --name=dz-test --namespace=mariadb-test --set securitySettings.enabled=false

$  oc get all -n mariadb-test
NAME                     DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
deploy/dz-test-mariadb   1         1         1            1           1m

NAME                            DESIRED   CURRENT   READY     AGE
rs/dz-test-mariadb-7457949548   1         1         1         1m

NAME                                  READY     STATUS    RESTARTS   AGE
po/dz-test-mariadb-7457949548-wlwbj   1/1       Running   0          1m

NAME                  TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)    AGE
svc/dz-test-mariadb   ClusterIP   172.30.92.234   <none>        3306/TCP   1m
```

File: stable/mariadb/Chart.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 name: mariadb
-version: 2.1.14
+version: 2.1.15
```
Comment (timstoop) [inline stable/mariadb/Chart.yaml:2]: For a new feature, the minor version should increase.

**Review by timstoop — CHANGES_REQUESTED** [review]

File: stable/mariadb/Chart.yaml (hunk 1 above)
Comment (sameersbn) [reply stable/mariadb/Chart.yaml:2]: @timstoop  I believe a minor version bump is not required here. We normally require that the minor versions be updated when there is a breaking change. This commit adds an optional parameter to turn off the `securityContext` configuration (enabled by default), so it should be fine if existing users upgrade to a chart version that has this change.

Comment (timstoop) [reply stable/mariadb/Chart.yaml:2]: Not to contradict you, but I've been told on other PRs that any new option would warrant a minor version increase. I'll bring it to the table in #3011, as it's probably good to have some guidelines!

**Pull request merged**